File: java/Euler003.java
import java.util.Date;

/*
 Project Euler Problem 3
 =======================

 The prime factors of 13195 are 5, 7, 13 and 29.

 What is the largest prime factor of the number 600851475143?
 */

public class Euler003 {
    public static void main(String[] args) {
        Date start, end;
        start = new Date();

        double number = 600851475143d;
        long largestPrime = 0;
        for (int i = 3; i < (int) Math.sqrt(number) + 1; i += 2) {

            if (number % i == 0) {
                if(isPrime((int)number/i)){
                    largestPrime = (int)number/i;
                    break;
                }
                if (isPrime(i) && i > largestPrime) {
                    largestPrime = i;
                }

            }
        }

        end = new Date();
        System.out.println(largestPrime);
        System.out.println("Execution Time: "
                + (end.getTime() - start.getTime()));
    }

    public static boolean isPrime(long num) {

        double sqrt = Math.sqrt(num);
        int sqrtCeil = (int) Math.ceil(sqrt) + 1;

        if (num == 0 || num == 1)
            return false;
        if (num == 2)
            return true;
        if (num % 2 == 0)
            return false;
        for (int i = 3; i < sqrtCeil; i += 2) {

            if ((num % i) == 0)
                return false;
        }
        return true;
    }
}

File: python/euler003.py
"""
 Project Euler Problem 3
 =======================

 The prime factors of 13195 are 5, 7, 13 and 29.

 What is the largest prime factor of the number 600851475143?
"""
from math import sqrt

def factorize(n):
    gf = 3
    while n % 2 == 0:
        n /=2
    for i in range(1, int(sqrt(n))):
        if n % i == 0  and isPrime(i):
            if i > gf:
                gf = i
            if isPrime(n /i):
                gf = n /i
                break
    return gf

def isPrime(n):
    if n == 1:
        return False
    elif n % 2 == 0 or n % 3==0:
        return False
    else:
        for i in range (5, int(sqrt(n)) + 1, 2):
            if n % i == 0:
                return False
        return True

print factorize(81)
